### ISSUE TITLE
Add more values to upsd input or make new one for NUT-ups

### ISSUE THREAD

**Issue by masterwishx** [influxdata/telegraf#14425]
### Use Case

There is much more [52 values] in NUT-ups ,that can be exported to InfluxDBv2, currently using https://github.com/jwillmer/nut-influxdbv2 as exporter to InfluxDBv2

![image](https://github.com/influxdata/telegraf/assets/28630321/49df9a85-c245-4f1b-ba8f-88e6482e5ce0)

in upsd input currenty only 18 values:

![image](https://github.com/influxdata/telegraf/assets/28630321/061b308e-637e-4a4d-94b5-f37df259b0d6)

### Expected behavior

.

### Actual behavior

.

**Comment by srebhan** [comment]
@masterwishx a PR to add those values is appreciated!

**Comment by masterwishx** [comment]
> PR to add those values is appreciated!

Do you mean the list of them ?
in the expoter im using now its getting automaticly if im not mistaken https://github.com/jwillmer/nut-influxdbv2/blob/12d8ca71e0b08f1c276ab0687aa91cde8c88a1ad/nut-influxdbv2-exporter.py#L157C9-L157C50

but in code here this is the static list ?https://github.com/influxdata/telegraf/blob/8c3bfab0c6a357a5173b01208cc9e8a586adbf74/plugins/inputs/upsd/upsd.go#L88

if you need only list of them i can make it

**Comment by masterwishx** [comment]
an Example :  

```
battery.capacity: 7.00
battery.charge: 100
battery.charge.low: 20
battery.charge.restart: 0
battery.protection: yes
battery.runtime: 3873
battery.runtime.low: 180
battery.type:
device.mfr: Eaton
device.model: unknown 2000
device.type: ups
driver.debug: 0
driver.flag.allow_killpower: 0
driver.name: usbhid-ups
driver.parameter.pollfreq: 30
driver.parameter.pollinterval: 2
driver.parameter.port: auto
driver.parameter.synchronous: auto
driver.state: updateinfo
driver.version: 2.8.1
driver.version.data: MGE HID 1.46
driver.version.internal: 0.52
driver.version.usb: libusb-1.0.26 (API: 0x1000108)
input.bypass.frequency: 49.9
input.bypass.voltage: 234.0
input.frequency: 49.9
input.frequency.nominal: 50
input.transfer.high: 300
input.transfer.low: 100
input.voltage: 234.0
input.voltage.nominal: 235
outlet.1.status: on
outlet.desc: Main Outlet
outlet.id: 0
outlet.switchable: no
output.current: 1.60
output.frequency: 49.9
output.frequency.nominal: 50
output.voltage: 229.0
output.voltage.nominal: 230
ups.beeper.status: enabled
ups.delay.shutdown: 20
ups.delay.start: 30
ups.firmware:
ups.load: 19
ups.load.high: 105
ups.mfr: Eaton
ups.model: unknown 2000
ups.power: 389
ups.power.nominal: 2000
ups.productid: ffff
ups.realpower: 253
ups.realpower.nominal: 1600
ups.shutdown: enabled
ups.start.auto: yes
ups.start.battery: yes
ups.start.reboot: yes
ups.status: OL
ups.temperature: 24.9
ups.test.interval: 604800
ups.test.result: Done and passed
ups.timer.shutdown: -1
ups.timer.start: -1
ups.type: online
ups.vendorid: 0463
```

**Comment by powersj** [comment]
@masterwishx, 

Based on that list, Telegraf already produces the following fields:

```text
battery.charge: 100
battery.runtime.low: 180
input.frequency: 49.9
input.transfer.high: 300
input.transfer.low: 100
input.voltage: 234.0
input.voltage.nominal: 235
ups.temperature: 24.9
ups.load: 19
ups.realpower.nominal: 1600
output.voltage: 229.0
ups.realpower: 253
ups.delay.shutdown: 20
ups.delay.start: 30
ups.status: OL
```

I would consider adding the following as floats:

```text
battery.capacity: 7.00
battery.runtime: 3873
input.bypass.frequency: 49.9
input.bypass.voltage: 234.0
input.frequency.nominal: 50
output.current: 1.60
output.frequency: 49.9
output.frequency.nominal: 50
output.voltage.nominal: 230
ups.power: 389
ups.power.nominal: 2000
```

The following as string fields:

```
battery.protection: yes
device.mfr: Eaton
device.model: unknown 2000
driver.version: 2.8.1
driver.version.data: MGE HID 1.46
driver.version.internal: 0.52
driver.version.usb: libusb-1.0.26 (API: 0x1000108)
ups.firmware:
ups.mfr: Eaton
ups.model: unknown 2000
ups.productid: ffff
ups.test.result: Done and passed
```

And skip on the following fields. Happy to hear if any of these are useful for generating any additional metrics or for monitoring:

```
battery.type:
battery.charge.low: 20
battery.charge.restart: 0
device.type: ups
driver.debug: 0
driver.flag.allow_killpower: 0
driver.name: usbhid-ups
driver.parameter.pollfreq: 30
driver.parameter.pollinterval: 2
driver.parameter.port: auto
driver.parameter.synchronous: auto
driver.state: updateinfo
outlet.1.status: on
outlet.desc: Main Outlet
outlet.id: 0
outlet.switchable: no
ups.beeper.status: enabled
ups.load.high: 105
ups.shutdown: enabled
ups.start.auto: yes
ups.start.battery: yes
ups.start.reboot: yes
ups.test.interval: 604800
ups.timer.shutdown: -1
ups.timer.start: -1
ups.type: online
ups.vendorid: 0463
```

**Comment by masterwishx** [comment]
> And skip on the following fields. Happy to hear if any of these are useful for generating any additional metrics or for monitoring:

im using some of these values from exporter that using now, also posted dashboard in grafana :   
https://grafana.com/grafana/dashboards/20077-nut-ups/

Like: 
ups.type: online
ups.vendorid: 0463
ups.test.interval: 604800 (1 week)
device.type: ups

Also i saw in other dashboards,maybe usefull to add :
ups.beeper.status: enabled 
ups.shutdown: enabled

![image](https://github.com/influxdata/telegraf/assets/28630321/2932eb9e-b194-4f6a-bcc4-7095a94ad517)

**Comment by masterwishx** [comment]
@powersj  Maybe will be good to add all the values if it possible ,then users can decide what they need ...

**Comment by powersj** [comment]
Hi @masterwishx 

Are you willing to put up a PR with the new fields?

> @powersj Maybe will be good to add all the values if it possible ,then users can decide what they need ...

In general I am not opposed to adding new ones, but I also want a field to be of value in terms of generating metrics. Fields like `outlet.switchable: no`, something that should not ever change, is something I would happily pass on. It is also unfortunate that the plugin currently defines every single field explicitly so adding everything is trying given the wide range of fields that could exists per vendor.

**Comment by masterwishx** [comment]
> Are you willing to put up a PR with the new fields?

I can try, very sad we can't add values automatically as exporter I'm using now...

**Comment by masterwishx** [comment]
> Are you willing to put up a PR with the new fields?

https://github.com/influxdata/telegraf/pull/14447 Please check it

**Comment by srebhan** [comment]
Fixed by #14447.
